pkg/service/backup/worker_schema.go

```diff
@@ -100,3 +125,51 @@ func (w *worker) UploadSchema(ctx context.Context, hosts []hostInfo) (stepError
 
 	return hostsInParallel(hostPerLocation, parallel.NoLimit, f, notify)
 }
+
+func marshalAndCompressSchema(schema query.DescribedSchema) (*bytes.Buffer, error) {
+	rawSchema, err := json.Marshal(schema)
+	if err != nil {
+		return nil, errors.Wrap(err, "marshal schema")
+	}
+
+	b := &bytes.Buffer{}
+	gw := gzip.NewWriter(b)
+	if _, err := gw.Write(rawSchema); err != nil {
+		return nil, errors.Wrap(err, "write compressed schema")
+	}
+	if err := gw.Close(); err != nil {
+		return nil, errors.Wrap(err, "close gzip writer")
+	}
+
+	return b, nil
+}
+
+// isDescribeSchemaSafe checks if restoring schema from DESCRIBE SCHEMA WITH INTERNALS is safe.
+func isDescribeSchemaSafe(ctx context.Context, client *scyllaclient.Client, hosts []string) (bool, error) {
+	const (
+		ConstraintOSS = ">= 6.0, < 2000"
+		ConstraintENT = ">= 2024.2, > 1000"
+	)
+
+	for _, h := range hosts {
+		ni, err := client.NodeInfo(ctx, h)
```

Comment:
NodeInfo on agent is not a single call to Scylla's API. The result is the concatenation of many different calls made on scylla server by manager-agent (see `pkg/cmd/agent/nodeinfo.go`).
It can be replaced with direct call to ScyllaAPI (through agent's proxy) -> "/storage_service/scylla_release_version"

Reply:
The same idea as for restore described in: https://github.com/scylladb/scylla-manager/pull/3876#discussion_r1641978810.